### PR TITLE
refactor: use config crate for tap-agent

### PR DESCRIPTION
**Pull request by gusinacio**

### CODE REVIEW

File: common/src/subgraph_client/client.rs [hunk 1]
```diff
@@ -88,12 +88,23 @@ impl DeploymentDetails {
...
+        Self::for_graph_node_url(
+            Url::parse(graph_node_status_url)?,
+            Url::parse(graph_node_base_url)?,
+            deployment,
+        )
+    }
+
+    pub fn for_graph_node_url(
+        graph_node_status_url: Url,
+        graph_node_base_url: Url,
```
Comment (suchapalaver) [inline common/src/subgraph_client/client.rs:101]: You don't want to pass those in as a struct so there's no mixing them up?

File: config/src/grt.rs [hunk 2]
```diff
@@ -4,10 +4,18 @@
 use bigdecimal::{BigDecimal, ToPrimitive};
 use serde::{de::Error, Deserialize};
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Default)]
 pub struct NonZeroGRT(u128);
 
 impl NonZeroGRT {
+    pub fn new(value: u128) -> Result<Self, String> {
+        if value == 0 {
```
Comment (suchapalaver) [inline config/src/grt.rs:12]: Have you ever used something like https://doc.rust-lang.org/stable/std/num/type.NonZeroU128.html for this?

**Review by suchapalaver — DISMISSED** [review]

File: common/src/subgraph_client/client.rs (hunk 1 above)
Comment (gusinacio) [reply common/src/subgraph_client/client.rs:101]: There's a second refactor going on, I want to change as little as possible in common.

File: config/src/grt.rs (hunk 2 above)
Comment (gusinacio) [reply config/src/grt.rs:12]: TIL, but this is used only in tests.

**Review by suchapalaver — APPROVED** [review]

**Pull request merged**